cpp/marc_tools/marc_grep_multiple.sh

```diff
@@ -22,25 +22,30 @@ for filename in "$@"; do
             found=0
 	fi
     else
-	tar_filename=${filename%.gz}
+    filebasename="/tmp/$(basename ${filename})"
```

Comment:
Variable names should be improved. This variable is called filebasename, but contains a full path. Also, there should be some kind of uuid so the script does not try to re-use an existing directory in case something goes wrong (you can e.g. use the uuidgen command).